patcher/src/services/session/sounds.ts

```diff
@@ -20,53 +20,49 @@ const UNMUTE_MUSIC = 'cse-patcher/sounds/UNMUTE_MUSIC';
 
 export interface SoundsAction extends BaseAction {}
 
-export function muteSounds(): SoundsAction {
-  localStorage.setItem(localStorageKey, JSON.stringify(true));
+export interface SoundsState {
+  playMusic: boolean;
+  playSound: boolean;
+}
+
+export function muteSounds(state: SoundsState): SoundsAction {
+  localStorage.setItem(localStorageKey, JSON.stringify(merge(state, {playSound: false})));
   return {
     type: MUTE_SOUNDS,
     when: new Date(),
   };
 }
 
-export function unMuteSounds(): SoundsAction {
-  localStorage.setItem(localStorageKey, JSON.stringify(false));
+export function unMuteSounds(state: SoundsState): SoundsAction {
+  localStorage.setItem(localStorageKey, JSON.stringify(merge(state, {playSound: true})));
   return {
     type: UNMUTE_SOUNDS,
     when: new Date(),
   };
 }
 
-export function muteMusic(): SoundsAction {
-  localStorage.setItem(localStorageKey, JSON.stringify(true));
+export function muteMusic(state: SoundsState): SoundsAction {
+  localStorage.setItem(localStorageKey, JSON.stringify(merge(state, {playMusic: false})));
   return {
     type: MUTE_MUSIC,
     when: new Date(),
   };
 }
 
-export function unMuteMusic(): SoundsAction {
-  localStorage.setItem(localStorageKey, JSON.stringify(false));
+export function unMuteMusic(state: SoundsState): SoundsAction {
+  localStorage.setItem(localStorageKey, JSON.stringify(merge(state, {playMusic: true})));
   return {
     type: UNMUTE_MUSIC,
     when: new Date(),
   };
 }
 
-export interface SoundsState {
-  playMusic: boolean;
-  playSound: boolean;
-}
-
-function getInitialState() {
-  return {
-    playMusic: true,
-    playSound: true,
-  };
-}
-
-const initialState = JSON.parse(localStorage.getItem(localStorageKey)) || false;
+const initialState = JSON.parse(localStorage.getItem(localStorageKey)) || {
+  playMusic: true,
+  playSound: true
+};
 
-export default function reducer(state: SoundsState = getInitialState(), action: SoundsAction = defaultAction): SoundsState {
+export default function reducer(state: SoundsState = initialState, action: SoundsAction = defaultAction): SoundsState {
```

Comment:
getInitialState should still be a function.  Also JSON.parse could through an exception if the data in local storage is not null but not valid, so need to safely try to parse that and catch any exceptions.